File: eureka/ios/nativeLib/nativeLib/Models/OpenWeatherStruct.swift
//
//  OpenWeatherStruct.swift
//  eureka
//
//  Created by work on 02/06/21.
//

import Foundation

public class OpenWeatherStruct {
  private var lat:Double = 0.0
  private var lon:Double = 0.0
  private var dt:Int64 = 0
  private var timezone:String = ""
  private var offset:Int64 = 0
  private var current:[String:Any] = [:]
  private var hourly:[[String:Any]] = []
  private var daily:[[String:Any]] = []
    
  public init() {}
  
  internal func getLat() -> Double {
    return lat
  }

  public func setLat(pLat:Double) {
    lat = pLat
  }

  internal func getLon() -> Double {
    return lon
  }

  public func setLon(pLon:Double) {
    lon = pLon
  }

  internal func getDt() -> Int64 {
    return dt
  }

  public func setDt(pDt:Int64) {
    dt = pDt
  }

  internal func getTimezone() -> String {
    return timezone
  }

  public func setTimezone(pTimezone:String) {
    timezone = pTimezone
  }

  internal func getOffset() -> Int64 {
    return offset
  }

  public func setOffset(pOffset:Int64) {
    offset = pOffset
  }

  public func getCurrent() -> [String:Any] {
    return current
  }

  public func setCurrent(pCurrent:[String:Any]) {
    current = pCurrent
  }

  public func getHourly() -> [[String:Any]] {
    return hourly
  }

  public func setHourly(pHourly:[[String:Any]]) {
    hourly = pHourly
  }

  public func getDaily() -> [[String:Any]] {
    return daily
  }

  public func setDaily(pDaily:[[String:Any]]) {
    daily = pDaily
  }
}

File: eureka/android/library/src/main/java/com/lifeplus/Pojo/OpenWeatherStruct.java
package com.lifeplus.Pojo;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.HashMap;

public class OpenWeatherStruct {
    @SerializedName("lat")
    private float lat;

    @SerializedName("lon")
    private float lon;

    @SerializedName("timezone")
    private String timezone;

    @SerializedName("timezone_offset")
    private long offset;

    @SerializedName("current")
    private HashMap<String, Object> current;

    @SerializedName("hourly")
    private ArrayList<HashMap<String, Object>> hourly;

    @SerializedName("daily")
    private ArrayList<HashMap<String, Object>> daily;

    public float getLat() {
        return lat;
    }

    public void setLat(float lat) {
        this.lat = lat;
    }

    public float getLon() {
        return lon;
    }

    public void setLon(float lon) {
        this.lon = lon;
    }

    public String getTimezone() {
        return timezone;
    }

    public void setTimezone(String timezone) {
        this.timezone = timezone;
    }

    public long getOffset() {
        return offset;
    }

    public void setOffset(long offset) {
        this.offset = offset;
    }

    public HashMap<String, Object> getCurrent() {
        return current;
    }

    public void setCurrent(HashMap<String, Object> current) {
        this.current = current;
    }

    public ArrayList<HashMap<String, Object>> getHourly() {
        return hourly;
    }

    public void setHourly(ArrayList<HashMap<String, Object>> hourly) {
        this.hourly = hourly;
    }

    public ArrayList<HashMap<String, Object>> getDaily() {
        return daily;
    }

    public void setDaily(ArrayList<HashMap<String, Object>> daily) {
        this.daily = daily;
    }
}